### PR TITLE
[Hub Generated] Review request for Microsoft.Devices to add version preview/2019-03-22-preview

### PR DESCRIPTION
**Pull request by reshantchandra**
If you are a MSFT employee you can view your [work branch via this link](https://portal.azure-devex-tools.com/branch/reshantchandra/azure-rest-api-specs/dev-iothub-Microsoft.Devices-2019-03-22-preview-06-13...master).

### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/azure-rest-api-specs#basics) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by sergey-shandar — CHANGES_REQUESTED** [review]
Can you please update the example to populate the value for new property?

**Review by shansin — APPROVED** [review]

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json [hunk 1]
```diff
@@ -2045,6 +2045,35 @@
             "name": "Capabilities",
             "modelAsString": true
           }
+        },
+        "locations": {
```
Comment (ravbhatnagar) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: how is the location of the primary different from location of the resource as provided via the top level property?

Comment (ravbhatnagar) [inline specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Is this the right way to model this concept? how do other Azure services do this? Can you provide some references, if you have? Should it be that primary is always defaulted to the location of the azure resource as provided by the customer and then the secondary is the closest fail over region?

Comment (ravbhatnagar) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Please share some more details on how these role types are used in iot hubs space. Feel free to set up a quick call to close on this.

**Review by ravbhatnagar — COMMENTED** [review]
@reshantchandra few notes. please take a look.

**Comment by reshantchandra** [comment]
closing this one will create a new one.

File: specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json (hunk 1 above)
Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: Location provided via top level property is immutable object referring to the original location (where resource is created).
while this primary location refers to the current location of the resource which can be different post fail over.

Comment (reshantchandra) [reply specification/iothub/resource-manager/Microsoft.Devices/preview/2019-03-22-preview/iothub.json]: As mentioned above these locations are used to determine the current location and the failover location of the resource. I have closed this PR and created a new one : https://github.com/Azure/azure-rest-api-specs/pull/6479 i have already added you to it. If you have further questions lets set up some time tomorrow.

**Pull request closed without merging**